src/lib/label/label-foundation.ts

```diff
@@ -0,0 +1,117 @@
+import { ICustomElementFoundation } from '@tylertech/forge-core';
+import { ILabelAdapter } from './label-adapter';
+import { LABEL_CONSTANTS } from './label-constants';
+
+export interface ILabelFoundation extends ICustomElementFoundation {
+  for: string | null | undefined;
+  forElement: HTMLElement | null | undefined;
+  freeze: boolean;
+  disconnect(): void;
+  update(): void;
+}
+
+export class LabelFoundation implements ILabelFoundation {
+  // State
+  private _for: string | null | undefined;
+  private _forElement: HTMLElement | null | undefined;
+  private _freeze = false;
```

Comment:
Let's make this "frozen" by default so that overhead of the `MutationObserver` is opt-in, and since it's relatively uncommon for labels to change since they are static in most usages by default.

I know we discussed this already, just putting it here as well in case anyone else chimes in.